Todo_list_dz/Todo_list.js

```diff
@@ -0,0 +1,108 @@
+
+let todoItemElems = [];
+
+function Task(description) {
+    this.description = description; // описание задачи
+    this.completed = false; // по умолчанию задача не выполнена (uncheck)
+}
+
+//создание шаблона для каждой отдельной задачи
+const createTemplate = (task,index) => {
+    return `
+    <div class = "todo-item ${task.completed ? 'checked' : ''}">
+        
+        <div class = "buttons">
+            <input onclick = "completeTask(${index})" class = "btn-complete" type = "checkbox" ${task.completed ? 'checked' : ''}></input>
+            </div>
+            <div class = "description"> ${task.description}</div>
+            <div class = "buttons">
+            <button onclick = "deleteTask(${index})" class = "btn-delete"> Delete </button>
+            <div class="date-task">${getUserTime()}</div> 
+        </div>
+    </div>
+    `
+}
+
+
+function getUserTime(t = new Date()) {
+    let Y = t.getFullYear();
+    let M = t.getMonth() + 1;
+    let D = t.getDate();
+
+    return `${Y}/${M}/${D}`
+}
+
+//фильтр для сбрасывания выполненных задач вниз
+const filterTask = () => {
+    const activeTasks = tasks.length && tasks.filter(item => item.completed == false);
+    const completedTasks = tasks.length && tasks.filter(item => item.completed == true);
+    tasks = [...activeTasks,...completedTasks];
+}
+
+//заполнение html списка задач
+const fillHtmlList = () => {
+todosWrapper.innerHTML = "";//зачищение данных
+//если массив tasks не пустой, то вытягиваю каждую задачу внутри массива перебором
+if(tasks.length > 0) { 
+    filterTask();
+    tasks.forEach((item, index) => {
+        todosWrapper.innerHTML += createTemplate(item, index);//+= к предыдущей итерации добавляем, то что на следующей итерации
+    });
+    todoItemElems = document.querySelectorAll('.todo-item');
+}
+}
+fillHtmlList();
+
+//обновление значений tasks
+const updateLocal = () => {
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+const completeTask = index => {
+    //console.log(index);
+    tasks[index].completed = !tasks[index].completed;//замена на противоположное значение true/false
+    if(tasks[index].completed) {
+        todoItemElems[index].classList.add('checked');
+    } else {
+        todoItemElems[index].classList.remove('checked');
+    }
+    updateLocal();
+    fillHtmlList();
+}
+
+addTaskBtn.addEventListener('click', () => {
+    tasks.push(new Task(descTaskInput.value));
+    updateLocal();
+    fillHtmlList();
+    descTaskInput.value = ""; // очистка input после добавления задачи, чтобы вводить новую в пустую строку
+})
+
+const deleteTask = index => {
+    todoItemElems[index].classList.add('delition')
+    setTimeout(() => {
+        tasks.splice(index, 1);
+        updateLocal();
+        fillHtmlList();
+},500)
+}
+
+
+delAllButton.addEventListener('click', () => {
+    todosWrapper.innerHTML = "";
+    tasks.splice(todosWrapper);
```

Comment:
не понял зачем это)
можно так
```suggestion
    tasks = [];
```